rest/oidc_api_test.go

```diff
@@ -0,0 +1,147 @@
+package rest
+
+import (
+	"io"
+	"net/http"
+	"net/http/httptest"
+	"net/url"
+	"strings"
+	"testing"
+
+	"github.com/couchbase/sync_gateway/auth"
+	"github.com/stretchr/testify/assert"
+	"github.com/stretchr/testify/require"
+)
+
+func TestGetOIDCCallbackURL(t *testing.T) {
+	var (
+		wellKnownPath           = "/.well-known/openid-configuration"
+		headerLocation          = "Location"
+		authProvider            = "provider"
+		authRedirectURI         = "redirect_uri"
+		providerNameGoogle      = "google"
+		providerNameSalesforce  = "salesforce"
+		clientIDGoogle          = "client.id.google"
+		clientIDSalesforce      = "client.id.salesforce"
+		validationKeySalesforce = "validation.key.salesforce"
+		validationKeyGoogle     = "validation.key.google"
+	)
+
+	// Mock OpenID Connect Provider Discovery Endpoints
+	var issuerGoogle, issuerSalesforce string
+	mux := http.NewServeMux()
+	mux.HandleFunc("/google/", func(res http.ResponseWriter, req *http.Request) {
+		if req.URL.Path != "/google/.well-known/openid-configuration" {
+			http.NotFound(res, req)
+			return
+		}
+		res.Header().Set("Content-Type", "application/json")
+		body := `{
+			"issuer": "${issuer}", 
+			"authorization_endpoint": "https://accounts.google.com/o/oauth2/v2/auth", 
+        	"token_endpoint": "https://oauth2.googleapis.com/token",
+			"jwks_uri": "https://www.googleapis.com/oauth2/v3/certs", 
+			"id_token_signing_alg_values_supported": ["RS256"]
+		}`
+		_, err := io.WriteString(res, strings.ReplaceAll(body, "${issuer}", issuerGoogle))
+		require.NoError(t, err, "Failed to write response body for google provider")
+	})
+	mux.HandleFunc("/salesforce/", func(res http.ResponseWriter, req *http.Request) {
+		if req.URL.Path != "/salesforce/.well-known/openid-configuration" {
+			http.NotFound(res, req)
+			return
+		}
```

Comment:
```suggestion
	mux.HandleFunc("/salesforce"+wellKnownPath, func(res http.ResponseWriter, req *http.Request) {
```

Reply:
This change has been included in latest revision.